include/seastar/parquet/exception.hh

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <exception>
+#include <sstream>
+
+namespace parquet {
+
+class parquet_exception : public std::exception {
+    std::string _msg;
+public:
+    ~parquet_exception() throw() override {}
+
+    static parquet_exception eof(const std::string& msg = "") {
+        std::stringstream ss;
+        ss << "Unexpected end of stream";
+        if (!msg.empty()) {
+            ss << ": " << msg << ".";
+        }
+        return parquet_exception(ss.str());
+    }
+
+    static parquet_exception nyi(const std::string& msg = "") {
```

Comment:
let's expand this name to `not_yet_implemented`, `nyi` is not a well-known acronym. Also - remove the default, as above and feel free to use seastar's `format()` instead of a raw string stream

Reply:
Thanks for pointing me to seastar::format. A format function is a blessing for writing error messages. I used it extensively in e23d38a.